
// Protocol buffer specification for Sentence.

syntax = "proto2";

import "ranker/preprocessing/feature.proto";

package proto;

message RankerDataset {
  repeated RankerDatapoint datapoint = 1;
}

message RankerDatapoint {
  // the word of the token
  optional string word = 1;
  // the word id in the embedding layer.
  optional int32 word_id = 2;
  // the top_k label hypothesis for this token. These represent the actions to be scored/ranked.
  // They come from the labeler.
  repeated Hypothesis hypotheses = 3;
  // the context features of the token. These are used while ranking the label hypothesis.
  optional ranker.Feature features = 4;
}

message Hypothesis {
  // the predicted label as a string.
  optional string label = 1;
  // the integer value of the predicted label based on the dep_label_enum.proto
  optional int32 label_id = 2;
  // the rank of this hypothesis.
  optional int32 rank = 3;
  // the reward for this hypothesis. This is determined based on the parser accuracy.
  optional float reward = 4;
}
